### DISCUSSION TITLE
pytorch 학습 모델 실행 관련

### DISCUSSION THREAD

**Discussion by Lee-JaeWon** [reinforcement-learning-kr/2021_RLKR_Drone_Delivery_Challenge_with_Unity#24] [General]
외부 python api에서 작성한 알고리즘을 통해 학습을 진행하여 얻은 pytorch의 학습 모델인 `.pt`파일을 (혹은 `.pth`)를 빌드된 유니티 환경인 `.exe` 에서 어떻게 테스트 할 수 있나요?
`.onnx`로 변환하여 유니티 에디터에서 테스트해야하나요?

++
baseline에서 `inference.py`를 참고해 해결중입니다.
`inference.py` 실행 시 오류가 나는데 주의해야할 점이 있나요?

`inference.py` 실행 시
`Agent_dqn.py`내의 아래와 같은 코드에서
```python
        def forward(self,x):
        
            front_obs, right_obs, rear_obs, left_obs, btn_obs, vector_obs = x
```
다음과 같은 오류가 발생합니다.
`ValueError: too many values to unpack (expected 6)`

**Comment by LeejwUniverse** [comment]
@Lee-JaeWon 님 안녕하세요. RL Drone Delivery challenge에 참여해주셔서 감사합니다.
inference.py를 통해 test를 진행해 보실 수 있습니다. 
제가 코드 테스트를 다시 한번 진행했는 데요. 현재 업로드된 코드에서 inference.py의 from Agent_dqn import Agent 로 import만 변경하시면, 바로 테스트 해보실 수 있습니다.

주말 동안 baseline code가 update된 부분들이 있어, 한번 확인해 주시면 감사하겠습니다.
추가로 문의 사항이 있으시면, comment 남겨 주시기 바랍니다.

감사합니다.

**Reply by Lee-JaeWon** [reply]
감사합니다. 수정된 baseline code를 반영해 해결하였습니다.
